### PR TITLE
Do not check IP source addresses when conf.checkIPsrc is False

### LINKED ISSUE #1894: conf.checkIPsrc Is not working as intended.
Greetings,

We believe we've found a bug, which still occurs in the latest version and can be seen on the master branch, where conf.checkIPsrc is not working properly. We've been running into issues where running the send receive function was not returning responses on packets sent to the destination. To us it appeared as if Scapy was not ignoring the inner packet source and that's when we found the conf.checkIPsrc option and proceeded to set it to 0 from its default of 1. We continued to get the same issue where we were not getting any responses except for the destination hop. We searched around in the source and we believe we've located a bug where you have src and dst mixed up in the following:

```
--- a/scapy/layers/inet.py
+++ b/scapy/layers/inet.py
@@ -726,8 +726,8 @@
     def answers(self, other):
         if not isinstance(other, IP):
             return 0
-        if not ( ((conf.checkIPsrc == 0) or (self.dst == other.dst)) and
-                 (self.src == other.src) and
+        if not ( ((conf.checkIPsrc == 0) or (self.src == other.src)) and
+                 (self.dst == other.dst) and
                  ( ((conf.checkIPID == 0)
                     or (self.id == other.id)
                     or (conf.checkIPID == 1 and self.id == socket.htons(other.id)))) and
```

After modifying the above locally within our container we no longer had any issues when passing conf.checkIPsrc=0.

Here's an example of running the sr function before we update the code:
```
>>> res,un=sr(IP(dst="4.2.2.2",ttl=2,flags='DF',id=RandShort())/ICMP(id=os.getpid()), timeout=2)
Begin emission:
.Finished sending 1 packets.
..............................
Received 31 packets, got 0 answers, remaining 1 packets
>>> res.summary()
>>> un.summary()
IP / ICMP 10.128.0.7 > 4.2.2.2 echo-request 0
>>> conf.checkIPsrc=0
>>> res,un=sr(IP(dst="4.2.2.2",ttl=2,flags='DF',id=RandShort())/ICMP(id=os.getpid()), timeout=2)
Begin emission:
Finished sending 1 packets.
..............................
Received 30 packets, got 0 answers, remaining 1 packets
>>> res.summary()
>>> un.summary()
IP / ICMP 10.128.0.7 > 4.2.2.2 echo-request 0
```

And after the fact:
```
>>> conf.checkIPsrc=0
>>> res,un=sr(IP(dst="4.2.2.2",ttl=2,flags='DF',id=RandShort())/ICMP(id=os.getpid()), timeout=2)
Begin emission:
.Finished sending 1 packets.
*
Received 2 packets, got 1 answers, remaining 0 packets
>>> res.summary()
IP / ICMP 10.128.0.7 > 4.2.2.2 echo-request 0 ==> IP / ICMP W.X.Y.Z > 10.128.0.7 time-exceeded ttl-zero-during-transit / IPerror / ICMPerror
>>> un.summary()
```

Would it be possible to have someone look into this and possibly update? Here's some additional info about the system I'm on:
```
# cat /etc/os-release
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.5.2
PRETTY_NAME="Alpine Linux v3.5"
HOME_URL="http://alpinelinux.org"
BUG_REPORT_URL="http://bugs.alpinelinux.org"

# python --version
Python 3.5.6
```
My Scapy version is Version 2.4.3rc1.dev7

Please let me know if any additional information is required.

### PR DESCRIPTION
**Pull request by guedou**
This PR fixes #1894.

I rewrote the whole test as it was quite difficult to understand.

### CODE REVIEW

File: scapy/layers/inet.py [hunk 1]
```diff
@@ -853,13 +853,21 @@ class IPerror(IP):
...
             return 0
-        if not (((conf.checkIPsrc == 0) or (self.dst == other.dst)) and
-                (self.src == other.src) and
-                (((conf.checkIPID == 0) or
-                  (self.id == other.id) or
-                  (conf.checkIPID == 1 and self.id == socket.htons(other.id)))) and  # noqa: E501
-                (self.proto == other.proto)):
+
+        # Check if IP addresses match
+        test_IPsrc = conf.checkIPsrc == 0 or self.src == other.src
```
Comment (p-l-) [inline scapy/layers/inet.py]: I'd rather have the boolean values tested as boolean:
```suggestion
        test_IPsrc = not conf.checkIPsrc or self.src == other.src
```

File: scapy/layers/inet.py [hunk 2]
```diff
@@ -853,13 +853,21 @@ class IPerror(IP):
...
+        test_IPdst = self.dst == other.dst
+
+        # Check if IP ids match
+        test_IPid = conf.checkIPID == 0 or self.id == other.id
```
Comment (p-l-) [inline scapy/layers/inet.py]:
```suggestion
        test_IPid = not conf.checkIPID or self.id == other.id
```

**Comment by guedou** [comment]
@p-l- thanks for the comments. I modified the patch.

**Comment by guedou** [comment]
@p-l- @gpotter2 ready to be merged!

**Pull request merged**